components/eamxx/src/diagnostics/tests/number_paths_tests.cpp

```diff
@@ -0,0 +1,216 @@
+        Field f(req.fid);
+        f.allocate_view();
+        f.get_header().get_tracking().update_time_stamp(t0);
+        input_fields.emplace(f.name(), f);
+      }
+      const auto &f = input_fields.at(req.fid.name());
+      diag->set_required_field(f.get_const());
+      REQUIRE_THROWS(diag->set_computed_field(f));
+    }
+    // Initialize the diagnostic
+    diag->initialize(t0, RunType::Initial);
+  }
+
+  // Run tests
+  {
+    // Construct random data to use for test
+    // Get views of input data and set to random values
+    const auto &qc_f = input_fields.at("qc");
+    const auto &qc_v = qc_f.get_view<Real **>();
+    const auto &qc_h = qc_f.get_view<Real **, Host>();
+    const auto &nc_f = input_fields.at("nc");
+    const auto &nc_v = nc_f.get_view<Real **>();
+    const auto &nc_h = nc_f.get_view<Real **, Host>();
+    const auto &qi_f = input_fields.at("qi");
+    const auto &qi_v = qi_f.get_view<Real **>();
+    const auto &qi_h = qi_f.get_view<Real **, Host>();
+    const auto &ni_f = input_fields.at("ni");
+    const auto &ni_v = ni_f.get_view<Real **>();
+    const auto &ni_h = ni_f.get_view<Real **, Host>();
+    const auto &qr_f = input_fields.at("qr");
+    const auto &qr_v = qr_f.get_view<Real **>();
+    const auto &qr_h = qr_f.get_view<Real **, Host>();
+    const auto &nr_f = input_fields.at("nr");
+    const auto &nr_v = nr_f.get_view<Real **>();
+    const auto &nr_h = nr_f.get_view<Real **, Host>();
+    const auto &pd_f = input_fields.at("pseudo_density");
+    const auto &pd_v = pd_f.get_view<Real **>();
+    const auto &pd_h = pd_f.get_view<Real **, Host>();
+    for(int icol = 0; icol < ncols; icol++) {
+      const auto &qc_sub = ekat::subview(qc_v, icol);
+      const auto &nc_sub = ekat::subview(nc_v, icol);
+      const auto &qi_sub = ekat::subview(qi_v, icol);
+      const auto &ni_sub = ekat::subview(ni_v, icol);
+      const auto &qr_sub = ekat::subview(qr_v, icol);
+      const auto &nr_sub = ekat::subview(nr_v, icol);
+      const auto &dp_sub = ekat::subview(pd_v, icol);
+      ekat::genRandArray(qc_sub, engine, pdf_qx);
+      ekat::genRandArray(nc_sub, engine, pdf_qx);
+      ekat::genRandArray(qi_sub, engine, pdf_qx);
+      ekat::genRandArray(ni_sub, engine, pdf_qx);
+      ekat::genRandArray(qr_sub, engine, pdf_qx);
+      ekat::genRandArray(nr_sub, engine, pdf_qx);
+      ekat::genRandArray(dp_sub, engine, pdf_pd);
+    }
+    // Grab views for each of the number path diagnostics
+    const auto &lnp   = diags["lnp"]->get_diagnostic();
+    const auto &inp   = diags["inp"]->get_diagnostic();
+    const auto &rnp   = diags["rnp"]->get_diagnostic();
+    const auto &lnp_h = lnp.get_view<const Real *, Host>();
+    const auto &inp_h = inp.get_view<const Real *, Host>();
+    const auto &rnp_h = rnp.get_view<const Real *, Host>();
+
+    for(const auto &dd : diags) {
+      dd.second->compute_diagnostic();
+    }
+    // test manual calculation vs one provided by diags
```

Comment:
Here too: before comparing the diags with the manually computed ones, you need to sync `inp`, `rnp`, `lnp` to host first.